File: app/src/main/java/com/vdovin/spacex/screen/detail/structure/LaunchDetailsView.java
package com.vdovin.spacex.screen.detail.structure;

import android.util.Pair;

import com.google.android.youtube.player.YouTubePlayer;
import com.vdovin.spacex.base.BaseView;
import com.vdovin.spacex.database.model.SpaceX;

import io.reactivex.Observable;

public interface LaunchDetailsView extends BaseView {

    void backPressed();

    void setupView(SpaceX spaceX);

    void openWiki(String wikiLink);

    Observable<Pair<YouTubePlayer, Boolean>> initializeYoutubePlayer();

    void playVideo(String youtubeId, YouTubePlayer youTubePlayer, boolean wasRestored);

    Observable<Boolean> detectFullscreen();

    Observable<Object> launchImageClicked();

    Observable<Object> backButtonClicked();

    Observable<Object> linkClicked();
}

File: app/src/main/java/com/vdovin/spacex/screen/details/LaunchDetailsView.kt
package com.vdovin.spacex.screen.details

import com.google.android.youtube.player.YouTubePlayer
import com.vdovin.spacex.base.BaseView

interface LaunchDetailsView : BaseView {

    fun backPressed()

    fun backButtonClicked(backClickListener: () -> Unit)

    fun launchImageClicked(launchImageListener: (YouTubePlayer, Boolean) -> Unit)

    fun linkClicked(linkListener: () -> Unit)

    fun setTitle(title : String)

    fun setDate(date : String)

    fun setDetails(details: String)

    fun setRocketName(rocketName: String)

    fun setPayloadMass(payloadMass: Double)

    fun setPayloadMassVisibility(visibility : Int)

    fun setLaunchImageVisibility(visibility: Int)

    fun setImage(imgSource: String)

    fun openWiki(wikiLink: String)

}